src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileParser.Cache.cs

```diff
@@ -0,0 +1,88 @@
+﻿extern alias __uno;
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+
+namespace Uno.UI.SourceGenerators.XamlGenerator
+{
+	internal partial class XamlFileParser
+	{
+		private struct CachedFileKey : IEquatable<CachedFileKey>
+		{
+			public CachedFileKey(string[] includeXamlNamespaces, string[] excludeXamlNamespaces, string file, int contentsHash)
+			{
+				IncludeXamlNamespaces = includeXamlNamespaces;
+				ExcludeXamlNamespaces = excludeXamlNamespaces;
+				File = file;
+				ContentsHash = contentsHash;
+			}
+
+			public string[] IncludeXamlNamespaces { get; }
+			public string[] ExcludeXamlNamespaces { get; }
+			public string File { get; }
+			public int ContentsHash { get; }
```

Comment:
Let's add the original file last modified time as well (if we can). I'm not sure if `AdditionalFile` supports that.

Reply:
There is no way to get that from `AdditionalFiles`.

I also think this can break the cache if a modification has been made then reverted. The last modified time will change but the contents are not and the cache should ideally be used.

Reply:
Interesting, then maybe that's when `AdditionalFiles` makes sense, since `AdditionalText` has a `GetChecksum` method we probably should be using it for the cache instead of building it manually.

Reply:
Interesting. Though I can't do that until `Uno.SourceGeneration` pushes a new version with `AdditionalFiles` support.

Reply:
Just figured out that Uno.SourceGeneration already pushed a dev version to azure feed (I was looking on NuGet.org). I'll prepare a separate PR to use AdditionalFiles and after that I'll get back to this

Reply:
I've just pushed the new build to nuget.org